### PR TITLE
[#4656] Revert PR #3633 so that mux expressions will be constant folded

### PR DESCRIPTION
**Pull request by kfcripps**
Reverts https://github.com/p4lang/p4c/commit/4a74084c675a3e8bdf7a318e468cbd5bb27c6914, as discussed as an alternate fix for #4656 in https://github.com/p4lang/p4c/pull/4657.

### CODE REVIEW

**Comment by ChrisDodd** [comment]
Looks like this is supceded by #4657, so this can be closed?

**Comment by kfcripps** [comment]
@ChrisDodd It is an alternate approach to #4657. I plan to close one after the other is merged.

**Comment by kfcripps** [comment]
As mentioned in #4657, I'm actually preferring this PR at this point.

File: testdata/p4_16_samples/issue4656_const_fold_generic_switch_label_expr.p4 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+extern bit<8> baz();
+action a(){}
+action b(){}
+control c() {
+    table t  {
+        actions = { a ; b;  }
+    }
+    apply {
+        switch(baz()) {
+            1 + 2 == 3 ? 1 : 2 : { foo(); }
```
Comment (vlstill) [inline testdata/p4_16_samples/issue4656_const_fold_generic_switch_label_expr.p4:14]: I am just surprised this is not syntactically ambiguous, the 2 `:` look quite weird :-). (This is not an issue with the PR/test, I'm just surprised the grammar allows it).

**Review by vlstill — APPROVED** [review]

File: testdata/p4_16_samples/issue4656_const_fold_generic_switch_label_expr.p4 (hunk 1 above)
Comment (jafingerhut) [reply testdata/p4_16_samples/issue4656_const_fold_generic_switch_label_expr.p4:14]: A single token of lookahead in Bison is apparently enough for it to figure it out, especially from the context that this is inside of a `switch` statement.

**Comment by fruffy** [comment]
https://github.com/p4lang/p4-spec/issues/1281 determines whether we may merge this particular PR, too.

**Comment by vlstill** [comment]
> superseded by https://github.com/p4lang/p4c/pull/4657

OK, so should this one be closed now?

**Comment by kfcripps** [comment]
Sure, it will need to be rebased even if the outcome of https://github.com/p4lang/p4-spec/issues/1281 is to remove the error.

**Pull request closed without merging**